Sideload examples in documentation

ronald-jaepel opened the pull request:
## Summary of changes

This PR introduces structured support for sideloading pre-run **RDM case studies** into the documentation build.
### RDM case study loading

A new pre-build script, `docs/source/load_studies.py`, is added. The script:

* clones or updates RDM project repositories via HTTPS
* checks out project and output refs
* reads `output/options.json` to determine the rendered source directory
* copies the study content into `docs/source/case_studies/<study>`
* updates the “Case Studies” toctree in `index.md` while preserving existing entries

---

### Separation of example types

The documentation right now distinguishes between:

* **Original examples** (`docs/source/examples/`)
  The old / original examples which are ultimately to be replaced by sideloaders.

* **Case studies** (`docs/source/case_studies/`)
  Precomputed snapshots copied from RDM output repositories.

---

### Case Studies navigation

The Case Studies section in `index.md` is updated by merging entries:

* existing manually maintained entries are preserved
* generated case studies are appended automatically
* duplicate entries are avoided

---

### MyST-NB execution configuration

The Sphinx configuration is updated to reflect the mixed execution model:

* `nb_execution_mode = "auto"` preserves per-notebook execution metadata
* `nb_execution_excludepatterns = ["case_studies/**"]` prevents execution of generated case studies

Existing notebooks that use cached execution continue to behave unchanged.

schmoelder commented:
After installing CADET-RDM from `main`, I still get the following error message:
```
Sphinx v8.2.3 in Verwendung
git@github.com:cadet/RDM-Example-Characterize-Chromatographic-System.git
Cloning git@github.com:cadet/RDM-Example-Characterize-Chromatographic-System.git into ../_tmp/study_root/Characterize-Chromatographic-System
Output repository was missing, cloning now.
Attempting to clone git@github.com:cadet/RDM-Example-Characterize-Chromatographic-System-Output.git into /home/jo/code/CADET-Process/docs/_tmp/study_root/Characterize-Chromatographic-System/output
Cloning git@github.com:cadet/RDM-Example-Characterize-Chromatographic-System-Output.git into /home/jo/code/CADET-Process/docs/_tmp/study_root/Characterize-Chromatographic-System/output
/home/jo/software/miniforge3/envs/cadet-process-docs/lib/python3.12/site-packages/cadetrdm/repositories.py:860: UserWarning: Repo version has outdated options hashes. Updating option hashes in output log.tsv.
  warnings.warn(
Found changes in these files:
 - ['log.tsv']
Commiting changes to repo /home/jo/code/CADET-Process/docs/_tmp/study_root/Characterize-Chromatographic-System/output

[main def98c0] Updated log hashes
 1 file changed, 4 insertions(+), 4 deletions(-)

Repo version 0.0.42 was outdated. Current CADET-RDM version is 0.1.7.
 Repo has been updated
Found changes in these files:
 - []
Commiting changes to repo /home/jo/code/CADET-Process/docs/_tmp/study_root/Characterize-Chromatographic-System

[main a978c1c] update cadetrdm version
 1 file changed, 2 insertions(+), 2 deletions(-)

Found matching study commit hash 0576c8d, but incorrect options hash (needs: 16t4x4h, has: f5ywb88)
No matching results were found for these options, but results with other options were found for this study versions. Did you recently change the options?
No results available for Case((PosixPath('/home/jo/code/CADET-Process/docs/_tmp/study_root/Characterize-Chromatographic-System'), '16t4x4h'))

Configuration error!

Versions
========

* Platform:         linux; (Linux-6.11.0-24-generic-x86_64-with-glibc2.39)
* Python version:   3.12.9 (CPython)
* Sphinx version:   8.2.3
* Docutils version: 0.21.2
* Jinja2 version:   3.1.6
* Pygments version: 2.19.1

Last Messages
=============

None.

Loaded Extensions
=================

None.

Traceback
=========

      File "/home/jo/software/miniforge3/envs/cadet-process-docs/lib/python3.12/site-packages/sphinx/config.py", line 616, in eval_config_file
        raise ConfigError(msg % traceback.format_exc()) from exc
    sphinx.errors.ConfigError: There is a programmable error in your configuration file:

    Traceback (most recent call last):
      File "/home/jo/software/miniforge3/envs/cadet-process-docs/lib/python3.12/site-packages/sphinx/config.py", line 601, in eval_config_file
        exec(code, namespace)  # NoQA: S102
        ^^^^^^^^^^^^^^^^^^^^^
      File "/home/jo/code/CADET-Process/docs/source/conf.py", line 262, in <module>
        load_study_results(name=name, study_url=study_url, study_branch=study_branch)
      File "/home/jo/code/CADET-Process/docs/source/conf.py", line 176, in load_study_results
        raise Exception(f"No results found for study {study_url}")
    Exception: No results found for study git@github.com:cadet/RDM-Example-Characterize-Chromatographic-System.git


The full traceback has been saved in:
/tmp/sphinx-err-dyif2e6u.log

To report this error to the developers, please open an issue at <https://github.com/sphinx-doc/sphinx/issues/>. Thanks!
Please also report this if it was a user error, so that a better error message can be provided next time.
```

hannahlanzrath commented:
For me currently every single one of those studies error with:

`File "/home/IBT/lanzrath/CADET/CADET-Process/docs/source/conf.py", line 176, in load_study_results
        raise Exception(f"No results found for study {study_url}")`

I will investigate.

It seems that the sideloader expects src files on the main branch, but the src files are saved within the respective study runs.

hannahlanzrath committed:
Fix test_initial_value in Test Optimization_Problem_LinCon

Co-authored-by: Johannes Schmölder <j.schmoelder@fz-juelich.de>

hannahlanzrath committed:
Docs: add load_study.py to sideload case studies

This commit introduces a way to sideload case studies in form of RDM
project repositories into the documentation.

Co-authored-by: r.jaepel <r.jaepel@fz-juelich.de>

hannahlanzrath committed:
Chore: version fix RDM to >= 1.1.1

hannahlanzrath committed:
Docs: add docs/_tmp to .gitignore

In docs/_tmp the Project and Output repositories are stored

hannahlanzrath committed:
Docs: add load_study.py script to RTD workflow

hannahlanzrath committed:
Docs: update conf.py

This commit changes the default notebook execution to auto, but excludes
the loaded in case studies, as these already have been executed before.
At the same time it conserves the load from cache behaviour for the old
examples.

hannahlanzrath commented:
Alright,

with these changes we can now load in examples from RDM Project repositories!

Katha and me will clean up the examples a little bit more, to make them presentable to the website. Additionally, I will try to fix some bugs in the older examples.

hannahlanzrath committed:
Docs: update secondary axes in old examples

hannahlanzrath committed:
Docs: update required purity in old examples

schmoelder committed:
Move matplotlib backend restoration to finally block

schmoelder committed:
Set show=False for internal plot calls

hannahlanzrath committed:
Docs: update SMA branch

hannahlanzrath committed:
Docs: add rectangular pulse example

hannahlanzrath committed:
Docs: Avoid double adding to toc tree for multi file examples

When an index file points to other files, they are added as
subdirectories instead of to the main tree.

hannahlanzrath committed:
Docs: Update output branches and add SMB
